crates/holochain_state/src/source_chain.rs

```diff
@@ -788,45 +790,74 @@ where
                         )",
                     });
                     sql.push_str(
-                        "
+                        format!("
                         )
                         AND
-                        (:entry_type IS NULL OR Action.entry_type = :entry_type)
+                        (:entry_type IS NULL OR Action.entry_type IN ({}))
                         AND
-                        (:action_type IS NULL OR Action.type = :action_type)
-                        ORDER BY Action.seq 
-                        ",
+                        (:action_type IS NULL OR Action.type IN ({}))
+                        ORDER BY Action.seq
+                        ", named_param_seq("entry_type", query.entry_type.as_ref().map_or(0, |t| t.len())), named_param_seq("action_type", query.action_type.as_ref().map_or(0, |t| t.len()))).as_str(),
                     );
-                    sql.push_str(if query.order_descending {" DESC"} else {" ASC"});
+                    sql.push_str(if query.order_descending { " DESC" } else { " ASC" });
                     let mut stmt = txn.prepare(&sql)?;
+
+                    let mut args: Vec<(String, Box<dyn rusqlite::ToSql>)> = Vec::with_capacity(6 + query.entry_type.as_ref().map_or(0, |t| t.len()) + query.action_type.as_ref().map_or(0, |t| t.len()));
```

Comment:
Code comment would be great here

Reply:
Also it could be clearer to just let-bind these two lengths so it's clear what they are from the name, especially since these same quantities are calculated twice

Reply:
Yes that would be better, updated